src/mame/vectorgraphic/v4_kbd.h

```diff
@@ -0,0 +1,63 @@
+// license:BSD-3-Clause
+// copyright-holders:Eric Anderson
+/**********************************************************************
+
+Vector 4 91-key keyboard
+
+**********************************************************************/
+
+#ifndef MAME_VECTORGRAPHIC_V4_KBD_H
+#define MAME_VECTORGRAPHIC_V4_KBD_H
+
+#pragma once
+
+#include "cpu/mcs48/mcs48.h"
+#include "sound/beep.h"
+
+
+class vector4_keyboard_device : public device_t
+{
+public:
+	// device type constructor
+	vector4_keyboard_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock = 0);
+
+	// callback configuration
+	auto txd_handler() { return m_txd_cb.bind(); }
+
+	// serial line input
+	DECLARE_WRITE_LINE_MEMBER(write_rxd);
+
+protected:
+	// device-level overrides
+	void device_resolve_objects() override;
+	void device_start() override;
+	ioport_constructor device_input_ports() const override;
+	void device_add_mconfig(machine_config &config) override;
+	const tiny_rom_entry *device_rom_region() const override;
+
+private:
+	// MCU handlers
+	uint8_t p1_r();
+	void p2_w(uint8_t data);
+	void leds_w(uint8_t data);
+
+	// address maps
+	void prog_map(address_map &map);
+	void ext_map(address_map &map);
+
+	// object finders
+	required_device<mcs48_cpu_device> m_mcu;
+	required_device<beep_device> m_beeper;
+	required_ioport_array<12> m_keys;
+	output_finder<1> m_leds;
+
```

Comment:
When you have a single output, you can use `output_finder<>` and not have to index it.